internal/ext/olm/constraints_test.go

```diff
@@ -0,0 +1,499 @@
+package olm
```

Comment:
nit: I like to use the `*_test` package to make sure we are testing the public facing interface of the package (and not sneaking some package protected changes